internal/thread/thread.go

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// +build !ebitensinglethread
```

Comment:
Instead of specifying a build tag here, I'd create `NoopThread`, rename `Thread` to `OSThread`, `New` to `NewOSThread`, and define `Thread` interface that includes `Loop` and `Call` function.